### ISSUE TITLE
Rare case of CKAN.ModuleNotFoundKraken that reports wrong mod in error message

### ISSUE THREAD

**Issue by nobodyhasthis** [KSP-CKAN/CKAN#1686]
**TL;DR** Issue raised to report a rare unhandled exception but believe it is NetKan related due odd dependency errors. This very hard to duplicate as it requires incorrect Metadata. As soon as that is fixed the problem disappears. However it was suggested that program behaviour should be reported here for later review. The larger issue of the reporting of a different mod to whatever's being installed is interesting. I believe this related to issue #1683 

--------------------------------------------------------------------------------------------------------------------------

**CKAN Version / KSP:**
CKAN 1.16.1-0-g2e91715
KSP 1.1.0 Pre-release build id = 01203 - 2016.04.12 at 22:50:28 CEST
Operating System: 
Windows 8.1
**The issue you are experiencing:**
Any attempt to install Kerbin Side Skyways will produce an unhandled exception. With error message related to Transfer Window Planner. Which seems like an unrelated mod. This may be due to the way I have installed other kerbin side mods. Traditionally during a manual install all mods are included in "complete" all in one mod version. However no feedback apart from the error. I am left to guess at what the root cause.
**How to recreate this issue:**
This is now out of date with recent metadata changes but included here anyway.
1. Use filter to search mods with "Kerbin Side" in name.
2. Install all of them except Ground Control and Skyways
3. Run Game
4. Install Kerbin Side Skyways to produce crash

optional step
5. Install Kerbin Side Ground Control which leads to file over write warning. Asking for another KSP-CKAN/CKAN-meta issue

CKAN error codes (if applicable): 
************** Exception Text **************
CKAN.ModuleNotFoundKraken: Cannot install TransferWindowPlanner, module not available
at CKAN.CkanModule.FromIDandVersion(IRegistryQuerier registry, String mod, KSPVersion ksp_version)
at CKAN.RelationshipResolver.c__AnonStorey0.<>m__0(String name)
at System.Linq.Enumerable.WhereSelectListIterator2.MoveNext() at System.Collections.Generic.List1..ctor(IEnumerable1 collection) at System.Linq.Enumerable.ToList[TSource](IEnumerable1 source)
at CKAN.RelationshipResolver..ctor(IEnumerable1 module_names, RelationshipResolverOptions options, IRegistryQuerier registry, KSPVersion kspversion) at CKAN.MainModList.ComputeConflictsFromModList(IRegistryQuerier registry, IEnumerable1 change_set, KSPVersion ksp_version)
at CKAN.Main.c__async1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at CKAN.Main.c__async0.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c__DisplayClass2.b__3(Object state)

Second Over Error when selection of Kerbin Side Ground Control
About to install...

Kerbin-Side Ground Control 0.3(cached)
Oh no! We tried to overwrite a file owned by another mod!
Please try a ckan update and try again.

If this problem re-occurs, then it maybe a packaging bug.
Please report it at:

https://github.com/KSP-CKAN/CKAN-meta/issues/new

Please including the following information in your report:

File : GameData/KerbinSide/Flags/alphastrikelogo.png
Installing Mod : KerbinSideGroundControl 0.3
Owning Mod : KerbinSide
CKAN Version : v1.16.1-0-g2e91715 (beta)

Your GameData has been returned to its original state.
Error!

**Comment by nobodyhasthis** [comment]
Was thinking this over today. I released that this unhandled exception could actually be very useful to keep in place. As it may indicate a desirable program behaviour. At the time the error occurred. I had a list of available mods ready for selection. They turned out in fact to be not compatible with KSP1.1. When this was fixed. Just before all of the metadata changes had gone through. I had downloaded the affected mods. Leaving CKAN in a temporary unstable state. Since initial report.

1. TWP has dropped off latest list of available mods as predicted.
2. The mutually exclusive check missing on Kerbin Side mods has started working. Blocking the user from selecting two conflicting mods.

If this unhandled exception error had not shown up. I would have still had transfer window planner installed via CKAN despite it not being compatible. This may indicate a weakness in the current system.

If a mod is available for one day. Then removed / changed to not compatible due to errors in supplied data. Does it still remain within the users list of installed mods anyway. With CKAN not expecting mods to roll back on version numbers or be removed from the index ?

**Comment by politas** [comment]
An installed mod will remain in the local metadata, and it seems that a removed mod that is not installed will only disappear on a CKAN restart after refreshing.

**Comment by Sakata-MC** [comment]
Not sure if something in the Community Tech Tree mod was updated since KSP has been fairly unstable for me the past few days, but noticed that today I start CKAN and I haven't any issue with selecting/deselecting any mods.
